pkg/agent/nodeportlocal/k8s/annotations.go

```diff
@@ -65,13 +70,16 @@ func assignPodAnnotation(pod *corev1.Pod, nodeIP string, containerPort, nodePort
 			klog.Warningf("Unable to unmarshal NodePortLocal annotation: %v", current[NPLAnnotationStr])
 		}
 
-		if !isNodePortInAnnotation(annotations, nodePort) {
-			annotations = append(annotations, NPLAnnotation{
-				PodPort:  containerPort,
-				NodeIP:   nodeIP,
-				NodePort: nodePort,
-			})
+		if isNodePortInAnnotation(annotations, nodePort, containerPort) {
+			// no updates required to the pod
+			return false
 		}
+
+		annotations = append(annotations, NPLAnnotation{
+			PodPort:  containerPort,
```

Comment:
I know it is not added by this PR, but why we do not name the field in the annotation as "PodPort", but not "ContainerPort" to be consistent with the K8s convention?

Reply:
Do you mean, change the annotation field to have "ContainerPort" instead of "PodPort"? If yes, can we take care of that once the other PR (which handles the service annotations gets merged), so that we  avoid any conflicts?

Reply:
Yes, that is what I mean. Another PR works.